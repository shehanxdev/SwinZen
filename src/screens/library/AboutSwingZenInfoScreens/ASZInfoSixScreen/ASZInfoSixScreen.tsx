import React from 'react';
import { View } from 'react-native';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { AboutSZInfoSixPoints, Color, TextAlignment, TextVariant } from '@sz/constants';

import { BaseLibraryScreen } from '../../components';

const TEST_ID_PREFIX = 'ASZInfoSixScreenTestID';

export function ASZInfoSixScreen() {
  return (
    <BaseLibraryScreen>
      <View style={tw`mx-4 mt-8`} testID={TEST_ID_PREFIX}>
        {AboutSZInfoSixPoints.map((data, index) => (
          <View key={`${TEST_ID_PREFIX}-Point-${index}`}>
            <Text color={Color.Neutral.White} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
              {data.title}
            </Text>
            <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
              {data.body}
              {`\n`}
            </Text>
          </View>
        ))}
      </View>
    </BaseLibraryScreen>
  );
}
