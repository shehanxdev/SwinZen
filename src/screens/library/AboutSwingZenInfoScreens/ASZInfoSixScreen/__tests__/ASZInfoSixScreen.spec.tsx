import { render } from '@testing-library/react-native';
import * as React from 'react';

import { ASZInfoSixScreen } from '../ASZInfoSixScreen';

describe('ASZInfoSix Screen', () => {
  const testID = 'ASZInfoSixScreenTestID';

  const getRenderedScreen = () => render(<ASZInfoSixScreen />);

  it(`should render ASZInfoSix Screen correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the ASZInfoSix Screen via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
