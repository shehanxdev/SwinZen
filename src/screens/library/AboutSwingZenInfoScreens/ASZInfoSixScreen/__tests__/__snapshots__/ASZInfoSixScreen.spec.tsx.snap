// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ASZInfoSix Screen should render ASZInfoSix Screen correctly 1`] = `
<BVLinearGradient
  colors={
    [
      4279917603,
      4278984207,
      4278650631,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "position": "relative",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../src/assets/images/grassBackground.jpg",
      }
    }
    style={
      {
        "opacity": 0.1,
        "position": "absolute",
      }
    }
  />
  <RCTSafeAreaView
    style={
      {
        "height": "100%",
        "paddingTop": 50,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
      keyboardShouldPersistTaps="handled"
    >
      <View>
        <View
          style={
            {
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 32,
            }
          }
          testID="ASZInfoSixScreenTestID"
        >
          <View>
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              Weight Transfer Reference Lines:
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              Static blue lines that are set to address ankle positions and head positions. Allows users to see how their weight transfer and head movement is happening throughout the swing.
              

            </Text>
          </View>
          <View>
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              Ball Position Line:
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              Static line at ball position. Shows ball position at address. The position will vary depending on the position of the camera. This is why your camera position should be directly in front of and square to the golfer.
              

            </Text>
          </View>
          <View>
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              Clubhead Tracking:
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              Shows the club path throughout the swing.
              

            </Text>
          </View>
          <View>
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              Head Position:
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              Dynamic white circle that shows the position of the head throughout the swing.
              

            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
